projects/go-lib/src/lib/components/go-header-bar/go-header-bar.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Location } from '@angular/common';
+import {Component, ElementRef, Input, OnInit, ViewChild} from '@angular/core';
+
+@Component({
+  selector: 'go-header-bar',
+  templateUrl: './go-header-bar.component.html',
+  styleUrls: ['./go-header-bar.component.scss']
+})
+export class GoHeaderBarComponent implements OnInit {
+
+  @Input() title: string;
+  @Input() showBackArrow: boolean = false;
+  @Input() goBackFn: Function;
+
+  @ViewChild('headerBar') headerBar: ElementRef;
+
+  constructor(private location: Location) {}
+
+  ngOnInit(): void {}
+
+  goBack(): void {
+    if (this.goBackFn) {
+      this.goBackFn();
+    } else {
+      this.location.back();
+    }
+  }
+
+  backButtonDisabled(): boolean {
+    return !this.goBackFn && window.history.length === 1;
+  }
+
+  getHeight(): string {
+    return this.headerBar.nativeElement.offsetHeight.toString() + 'px';
+  }
+
+  getWidth(): string {
+    return this.headerBar.nativeElement.parentElement.parentElement.offsetWidth.toString() + 'px';
```

Comment:
While this does work, it assumes a specific structure in the HTML. This will likely be prone to bugs if the implementor does not put the `<go-header-bar>` at the root level inside of the routes component. 

This works:
```
<go-header-bar>
</go-header-bar>

<div class="go-container">
  page content
</div>
```

This will break:
```
<div>
  <go-header-bar>
  </go-header-bar>
</div>

<div class="go-container">
  page content
</div>
```

Reply:
I injected the `ElementRef` but there is still a styling issue when wrapping in a div:

```html
<go-header-bar>
</go-header-bar>
```

Results in correct styling:
![image](https://user-images.githubusercontent.com/2780584/70347005-988a0b00-182d-11ea-80db-38a933d0db2b.png)


```html
<div>
    <go-header-bar>
    </go-header-bar>
</div>
```

results in incorrect style:
![image](https://user-images.githubusercontent.com/2780584/70347092-c5d6b900-182d-11ea-8b47-b6697b2200c2.png)

Reply:
Hmm, we might have to go with a slightly different approach. We could query for the closest element with the `go-layout__route-container-outlet` class. This would look like:

```
getWidth(): string {
  return this.elemRef.nativeElement.closest('.go-layout__route-container-outlet').offsetWidth.toString() + 'px';
}
```

Reply:
Additionally, we'll need may need to recalculate this width based on window resize. Angular might already know to call these methods based so you'll want to double check that.

Reply:
Another approach I was thinking of was to use a [portal](https://material.angular.io/cdk/portal/overview) to render this into the right slot in the layout. That way regardless of where in the template you put this, it will render in a place that we already know the dimensions and we don't have to do all these calculations.

Reply:
`this.elemRef.nativeElement.closest('.go-layout__route-container-outlet');` returns `null`. would it be a problem to query for it locally?